### PR TITLE
[GAWB-2156] Cannot Launch Analysis on entity of type X if its ID is hidden in the Data tab

### PR DESCRIPTION
**Pull request by rtitle**
See: https://broadinstitute.atlassian.net/browse/GAWB-2156

Summary:
- ~~Made the `entity-column` of `EntityTable` non reorderable. This means it can't be hidden via the Filter Columns dialog.~~
  - Update: This caused some undesired side effects to column reordering so I reverted it. It seems like the `:reorderable?` flag on table columns is a bit buggy.
- In the Launch Analysis modal, made it so you can click _anywhere_ in the table row to select an entity.
- In the Launch Analysis modal, made the entity column not a link. 
---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by rtitle** [comment]
@MattPutnam want to take a look at this for me? Thanks.

File: src/cljs/main/broadfcui/common/table_utils.cljs [hunk 1]
```diff
@@ -148,9 +148,12 @@
...
         (map-indexed
           (fn [row-index row]
-            (let [row-style (:row-style props)
-                  row-style (if (fn? row-style) (row-style row-index row) row-style)]
-              [:div {:style (merge {:display "flex" :alignItems "center"} row-style)}
+            (let [get-row-style #(if (fn? %) (% row-index row) %)
+                  row-style (-> props (:row-style) (get-row-style) (merge {:display "flex" :alignItems "center"}))
+                  div-props (merge {:style row-style}
+                                   (when-let [on-row-click (:on-row-click props)]
+                                     {:onMouseDown (fn [] (on-row-click (first row)))}))]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/table_utils.cljs]: `:on-row-click` should take the whole row, and probably the index, like row-style does.

File: src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs [hunk 2]
```diff
@@ -60,14 +60,12 @@
...
                       :else "#fff")})
-      :entity-name-renderer (fn [e]
-                              (let [entity-id (entity->id e)]
-                                (style/create-link
-                                  {:text (:name entity-id)
-                                   :onClick #(swap! state assoc
-                                                    :selected-entity entity-id
-                                                    :workflow-count (common/count-workflows
-                                                                      e (:root-entity-type props)))})))}]]
+      :on-row-click (fn [e]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs]: `e` is for event objects, so call this `row`.

**Review by MattPutnam — CHANGES_REQUESTED** [review]

**Comment by rtitle** [comment]
Thanks @MattPutnam, pushed a commit with changes.

File: src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs [hunk 3]
```diff
@@ -60,14 +60,13 @@
                     (cond (= (entity->id (first row-data)) (:selected-entity @state)) "yellow"
                       (even? row-index) (:background-light style/colors)
                       :else "#fff")})
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs]: Stick a `:cursor "pointer"` in here so rows look clickable

File: src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs [hunk 4]
```diff
@@ -60,14 +60,13 @@
...
-                              (let [entity-id (entity->id e)]
-                                (style/create-link
-                                  {:text (:name entity-id)
-                                   :onClick #(swap! state assoc
-                                                    :selected-entity entity-id
-                                                    :workflow-count (common/count-workflows
-                                                                      e (:root-entity-type props)))})))}]]
+      :on-row-click (fn [row-index row-data]
+                      (let [row (first row-data)
+                            entity-id (entity->id row)]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs]: Rename `row` to `entity`, and `row-data` to `row`.  You can also rename `row-index` to `_` to indicate it's unused (and it gets rid of an editor warning for me at least).

File: src/cljs/main/broadfcui/common/table_utils.cljs [hunk 5]
```diff
@@ -148,9 +148,12 @@
        [:div {:style (merge {:fontSize "80%" :fontWeight 500} (:body-style props))}
         (map-indexed
           (fn [row-index row]
-            (let [row-style (:row-style props)
-                  row-style (if (fn? row-style) (row-style row-index row) row-style)]
-              [:div {:style (merge {:display "flex" :alignItems "center"} row-style)}
+            (let [get-row-style #(if (fn? %) (% row-index row) %)
+                  row-style (-> props (:row-style) (get-row-style) (merge {:display "flex" :alignItems "center"}))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/table_utils.cljs]: Minor style thing: you can omit the parens around single entries in a threading macro.  So just `(-> props :row-style get-row-style ...)`.

**Review by MattPutnam — APPROVED** [review]
👍 with some cosmetic comments

File: src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs (hunk 3 above)
Comment (rtitle) [reply src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs]: Neat

**Comment by rtitle** [comment]
Made cosmetic changes

**Pull request merged**